### DISCUSSION TITLE
Kimai 2.2.1, LDAP install fails

### DISCUSSION THREAD

**Discussion by Leoskari** [kimai/kimai#4428] [Questions & Answers] [answered]
> Ubuntu 22.04.3 LTS

>  PHP 8.1.25 (cli) (built: Oct 27 2023 14:00:40) (NTS)
> 
> Copyright (c) The PHP Group
> Zend Engine v4.1.25, Copyright (c) Zend Technologies
>     with Zend OPcache v8.1.25, Copyright (c), by Zend Technologies

`apt-get install php-ldap`

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
php-ldap is already the newest version (2:8.2+93+ubuntu22.04.1+deb.sury.org+2).
0 upgraded, 0 newly installed, 0 to remove and 5 not upgraded.

`composer require laminas/laminas-ldap --optimize-autoloader -n`

Cannot use laminas/laminas-ldap's latest version 2.17.0 as it requires ext-ldap * which is missing from your platform.

In PackageDiscoveryTrait.php line 318:
                                                                                                                          
  Package laminas/laminas-ldap has requirements incompatible with your PHP version, PHP extensions and Composer version:  
    - laminas/laminas-ldap 2.17.0 requires ext-ldap * but it is not present. 

`composer why-not laminas/laminas-ldap 2.17.0`

In BaseDependencyCommand.php line 100:
                                                                 
Could not find package "laminas/laminas-ldap" in your project  

**So what im doing wrong?**

**Comment by kevinpapst** [comment]
> **So what im doing wrong?** 

You don't RTFM: https://www.kimai.org/documentation/ldap.html

<img width="1030" alt="Bildschirmfoto 2023-11-12 um 11 25 02" src="https://github.com/kimai/kimai/assets/533162/1c23dfea-4dcd-4121-8ae6-518ca66c24c0">

**Reply by Leoskari** [reply]
Its allready installed, as my first post show: "php-ldap is already the newest version", so whats wrong?

**Reply by kevinpapst** [reply]
This has absolutely nothing todo with Kimai. But I added another sentence to the docs anyway:

> FPM and CLI PHP use different configs, so installing might not be enough: use php -m to verify that the module is really loaded.

**Reply by Leoskari** [reply, marked as answer]
Hi, yes thats true, its not at all Kimai's fault. At this case with ubuntu 22.04 and php8.1 solution was:

`sudo apt-get install php8.1-ldap`

if you want to post it to docs

**Reply by kevinpapst** [reply]
okay, first time is always new. thanks for sharing, added it to the docs
